Closes #19547: Add Sync button for DataSource ListView

pheus opened the pull request:
### Fixes: #19547

### Summary
Adds a **Sync** button to the Data Sources list view so a sync can be triggered directly from the table (no need to open the detail page).

### Details
- Introduces a per‑row **Sync** action in the list view.
- Visibility/enabled state respects user permissions and current job status.
- Reuses the existing sync mechanism/queue; no new API surface.

### UX
- Button sits alongside existing row actions.
- Disabled while a sync is pending or running.

### Notes
Open to feedback on approach.

jeremystretch commented on netbox/core/views.py:
```diff
@@ -85,6 +85,8 @@
             _("Queued job #{id} to sync {datasource}").format(id=job.pk, datasource=datasource)
         )
+        if return_url := request.POST.get('return_url'):
+            return redirect(return_url)
```
The `return_url` value needs to be sanitized. We can have the view inherit [`GetReturnURLMixin`](https://github.com/netbox-community/netbox/blob/8311f457b5474a57fc31a5583a390c9404a3acd7/netbox/utilities/views.py#L147) and call `get_return_url()` to handle this.

Note: Even after making this change, CodeQL might still flag this. There are numerous similar false positives we're working on cleaning up.

pheus commented on netbox/core/tables/template_code.py:
```diff
@@ -26,3 +26,21 @@
+      </button>
+    {% else %}
+      <span class="inline-block" tabindex="0" data-bs-toggle="tooltip" data-bs-delay="100" data-bs-placement="bottom">
+        <button class="btn btn-primary" disabled>
```
Good catch! I've added `btn-sm` for consistent. Thanks!

pheus commented on netbox/core/tables/template_code.py:
```diff
@@ -26,3 +26,21 @@
+        <i class="mdi mdi-sync" aria-hidden="true"></i> {% trans "Sync" %}
+      </button>
+    {% else %}
+      <span class="inline-block" tabindex="0" data-bs-toggle="tooltip" data-bs-delay="100" data-bs-placement="bottom">
```
You’re right, the wrapper isn’t needed here. I’ve removed the `<span>` and the unused tooltip attributes.

On the markup, I can either keep two explicit button tags (clearer) or collapse into one with conditional attributes (less duplication). I have a slight preference for the two‑tag version for readability, but happy to switch if you prefer the single‑tag approach. Do you have a preference?

**Option A: two explicit buttons (readable)**

```python
DATA_SOURCE_SYNC_BUTTON = """
{% load helpers %}
{% load i18n %}
{% if perms.core.sync_datasource %}
    {% if record.ready_for_sync %}
        <button class="btn btn-primary btn-sm" type="submit" formaction="{% url 'core:datasource_sync' pk=record.pk %}?return_url={{ request.get_full_path|urlencode }}" formmethod="post">
            <i class="mdi mdi-sync" aria-hidden="true"></i> {% trans "Sync" %}
        </button>
    {% else %}
        <button class="btn btn-primary btn-sm" disabled>
            <i class="mdi mdi-sync" aria-hidden="true"></i> {% trans "Sync" %}
        </button>
    {% endif %}
{% endif %}
"""
```

**Option B: single button with conditional attributes (compact)**


```python
DATA_SOURCE_SYNC_BUTTON = """
{% load helpers %}
{% load i18n %}
{% if perms.core.sync_datasource %}
    <button class="btn btn-primary btn-sm" {% if record.ready_for_sync %}type="submit" formaction="{% url 'core:datasource_sync' pk=record.pk %}?return_url={{ request.get_full_path|urlencode }}" formmethod="post"{% else %}disabled{% endif %}>
        <i class="mdi mdi-sync" aria-hidden="true"></i> {% trans "Sync" %}
    </button>
{% endif %}
"""
```

pheus committed:
feat(core): Add Sync button for DataSource actions

Introduces a sync button in the DataSource table for improved user
interaction. Enables users to trigger sync actions directly from the
table, with context-sensitive availability based on permissions and
record status.

Closes #19547

pheus commented:
Thank you for taking the time to review this PR. I appreciate the thoughtful feedback and attention to detail.
